### ISSUE TITLE
Question: always_skip unless called explicitly

### ISSUE THREAD

**Issue by hectorv** [pre-commit/pre-commit#719]
Is there a way to add a hook to `.pre-commit-config.yaml` but have it always skip unless it's called explicitly? Granted, this is a weird use-case where I'm basically just trying to benefit from `pre-commit`'s package management abilities :)
So for example, while doing `git commit` or `pre-commit run` it would be marked as `Skipped` but would run when called like this:
```
pre-commit run <hook>
```

For context, I'm trying to simplify setting up [prmd](https://github.com/interagent/prmd) (a Ruby package) on Python repos. This already works (as part of a script)
`pre-commit run --verbose prmd --files <json>`
where
```
-   repo: /path/to/mirrors-prmd
    sha: v0.11.7
    hooks:
    -   id: prmd
        entry: prmd render
        args:
        - --template=main/api/doc/templates
```
but now I would like to prevent this hook from running in all other instances :)
As I said, weird use-case... maybe I just need to go to sleep now 😴

**Comment by asottile** [comment]
Totally get this usecase 😆 

I think there should be a better way to do this but I'll tell you how I currently do this and give some additional ideas and then a feature idea :)

### abusing hook stages

Note: I personally don't use pre-push so if I don't want a hook to run normally I configure it as follows:

```yaml
    -   id: dont-run-normally
        stages: [push]
```

That way during `commit` it doesn't normally run.  However when I want to manually run it I can do: `pre-commit run dont-run-normally --hook-stage push` to make it run.

(woops! turns out `--hook-stage` is undocumented on [pre-commit.com](https://pre-commit.com/#pre-commit-run)!)

### try-repo

Another idea is to use ]`pre-commit try-repo /path/to/repo`](https://pre-commit.com/#pre-commit-try-repo), probably not the best when you have to remember the incantation each time though :)

### rubyvenv

For ruby specifically, I've been slowly (haven't really made progress in a long time) been trying to factor out the pre-commit ruby support into a separate package (also so I can attempt to tackle #201 (the oldest as-of-now open issue)).  There's an incomplete, but for a lot of usecases works fine "ruby virtualenv"ish tool: https://github.com/asottile/rubyvenv -- it needs a lot of polishing, needs to support _building_ ruby, and still needs windows support somehow but otherwise is functional with prebuilt rubies.

### how I think this should work -- abusing hook stages for greater good!

I'd like to make the following work, does this sound reasonable?

the config:
```yaml
    -   id: dont-run-normally
        stages: [manual]
```

the invocation: `pre-commit run dont-run-normally --hook-stage manual`

thoughts?

**Comment by hectorv** [comment]
Thanks @asottile !
I like the idea of `stages: [manual]` 💯

**Comment by asottile** [comment]
want to try out the patch in #720 and see if it fits your needs?

**Comment by hectorv** [comment]
@asottile looks great! 💥 

```
❯ pip install -e git://github.com/pre-commit/pre-commit.git@manual#egg=pre-commit

❯ pre-commit run -h
usage: pre-commit run [-h] [--color {auto,always,never}] [-c CONFIG]
                      [--verbose] [--origin ORIGIN] [--source SOURCE]
                      [--commit-msg-filename COMMIT_MSG_FILENAME]
                      [--hook-stage {commit,commit-msg,manual,push}]
                      [--show-diff-on-failure]
                      [--all-files | --files [FILES [FILES ...]]]
                      [hook]
[...]
  --hook-stage {commit,commit-msg,manual,push}
                        The stage during which the hook is fired. One of
                        commit, commit-msg, manual, push
[...]

❯ pre-commit run --hook-stage manual prmd
prmd.....................................................................Passed
```

and silence for other stages 😃

**Comment by asottile** [comment]
sweet!

**Comment by asottile** [comment]
Thanks for the cool idea, this should land in the next _minor_ release :)  I have a few more things I want to pull into that before releasing.  For now the `stages: [manual]` bit is a valid 1.7.0 config so you can still commit that value, it just won't do anything (and until 1.8.0 (or install-from-git) it won't be runnable).

**Comment by asottile** [comment]
This has been released as part of [v1.8.0](https://github.com/pre-commit/pre-commit/releases/tag/v1.8.0)!
